plugin/helper/parser.go

```diff
@@ -34,6 +35,14 @@ func (c ParserConfig) Build(context plugin.BuildContext) (ParserPlugin, error) {
 		c.ParseTo.FieldInterface = entry.NewRecordField()
 	}
 
+	if reflect.DeepEqual(c.ParseFrom, c.ParseTo) && c.Preserve {
```

Comment:
Rather than using reflection here, can we instead do `c.ParseFrom.String() == c.ParseTo.String()`?

Reply:
Definitely. That's a much better idea